src/cpp/session/modules/automation/SessionAutomation.R

```diff
@@ -283,6 +283,9 @@
    envVars[rstudioEnvVars] <- list(NULL)
    envVars["R_SESSION_TMPDIR"] <- list(NULL)
    
+   # Make sure the automation server uses the same R session executable.
+   envVars[["RSTUDIO_WHICH_R"]] <- ps::ps_exe()
```

Comment:
@gtritchie this should fix the issue you saw with automation tests using the wrong rsession.